jsonstore/jsonstore.go

```diff
@@ -235,6 +247,35 @@ func (app *JSONStoreApplication) DeliverTx(tx []byte) types.ResponseDeliverTx {
 		}
 
 		break
+
+	case "upvoteValidator":
+		entity := body["entity"].(map[string]interface{})
+
+		// validate user exists
+		pubKeyBytes, errDecode := base64.StdEncoding.DecodeString(message["publicKey"].(string))
+
+		if errDecode != nil {
+			panic(errDecode)
+		}
+
+		publicKey := strings.ToUpper(byteToHex(pubKeyBytes))
+
+		var user User
+		err := db.C("users").Find(bson.M{"publicKey": publicKey}).One(&user)
+		if err != nil {
+			panic(err)
+		}
+
+		fmt.Println("user validated!")
+
+		// validate validator exists & update votes
+		validatorID := bson.ObjectIdHex(entity["validator"].(string))
+		err = db.C("validators").Update(bson.M{"_id": validatorID}, bson.M{"$addToSet": bson.M{"votes": user.ID}})
+		if err != nil {
```

Comment:
We are doing only `$addToSet` here. What about the scenario when the user wants to retract the vote assigned to a particular person?

Reply:
Thought of that but I didnt implement it as I wanted to test what I had first and was struggling with that at the end.

Reply:
I've added scenario to downvote validator.